src/lxml/html/clean.py

```diff
@@ -242,6 +243,9 @@ def __init__(self, **kw):
                                  "allow_tags and remove_unknown_tags")
             self.remove_unknown_tags = False
 
+        if not isinstance(self.host_whitelist, collection_types):
+            raise TypeError("host_whitelist must be a collection type")
```

Comment:
I think it would be nice to always turn the value into a frozenset (if non-empty).

Reply:
As defensive copying, and for speed of lookups?

Reply:
Both, yes. We should generally own config collections that the user passed into the constructor, and not let future modifications leak in. I'm aware that that wasn't done before.

Reply:
Hmm, just noticed that you said non-empty -- why so?

Reply:
What do you think -- any further changes needed?